products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,114 @@
+
+### Code Location
+
+_The path of the source code in the repository._
+
+### Testing Plan
+
+_How you will verify the behavior of your system. Once the system is written, this section should be updated to reflect the current state of testing and future aspirations._
+
+#### Testing the search input
+
+#### Testing the crawler
+
+### Logging
+
+By logging the term(s) that users are querying, we could theoretically identify trends to help increase the quality of our search results.
+
+### Debugging
+
+_How users can debug interactions with your system. When designing a system it's important to think about what tools you can provide to make debugging problems easier. Sometimes it's unclear whether the problem is in your system at all, so a mechanism for isolating a particular interaction and examining it to see if your system behaved as expected is very valuable. Once a system is in use, this is a great place to put tips and recipes for debugging. If this section grows too large, the mechanisms can be summarized here and individual tips can be moved to another document._
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+The search input would need to be sanitized.
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
+
+### Privacy Concerns
+
+Since the initial implementation will be an unauthenticated landing page, we will have little, if any, data about the user.
+
+While there shouldn't be any PII in our public documentation sources, it is possible it exists, and hasn't been discovered yet. Therefore, it is possible that our search results could lead public users to sources that contain PII.
+
+### Open Questions and Risks
+
+- Should we index private documentation sources? If so, what private documentation sources can we expose to public users, if any?
+- Should the landing page be public only, or should it also support authenticated users?
+- How often should we reindex our documentation sources? For perspective, Algolia's open source crawler runs every 24 hours.
+- What should the URL of the documentation landing page be? Since we have `design.va.gov`, does something like `docs.va.gov` make sense?
```

Comment:
This site is intended to be very product-team focused. I suspect there would be little, if any, value to a veteran that stumbled across the site we're proposing. 

That being said, I feel like the same argument could be made about `design.va.gov`, and that seemingly wasn't a strong enough argument to prevent `design.va.gov` from going live. 

Additionally, during a conversation with @rianfowler, he pointed out that he would prefer a route (I suspect something like `va.gov/docs`) over a subdomain because subdomains have additional challenges (specifically CORS) that routes don't.

Reply:
Yeah, I don't have a strong opinion either way. In any case, as long as we explicitly state what the site is _for_ when you get there, I'm okay with mostly whatever.

For example: https://developer.va.gov
It's a pretty generic subdomain, but once you go there, you can see it's clearly "A Veteran-centered API platform for securely accessing VA data." (Shout out to the awesome lighthouse team working on this! :raised_hands: )

I'm curious if anybody has a name with a bit more baked-in context, though.

Reply:
What is the process for getting a `*.va.gov` or `va.gov/*` URL? If that is our recommendation we should include that process here. I am on board with `docs.va.gov` or `va.gov/docs` but am not sold on it being necessary for MVP if the process is extensive.

If we are considering a va.gov URL, we will also need to consult with @mnorthuis as the guardian of va.gov URLs.

Reply:
Its definitely technically easier/faster to use a path instead of getting a new subdomain. I personally haven't gone through getting a new subdomain but there is some paperwork and a group within the VA that needs to approve it and then the DNS folks also need to be involved.. We would also need to setup SSL certs and the like.